src/core/mount.c

```diff
@@ -1883,7 +1884,11 @@ static int mount_load_proc_self_mountinfo(Manager *m, bool set_flags) {
                 if (!device || !path)
                         continue;
 
-                device_found_node(m, device, DEVICE_FOUND_MOUNT, DEVICE_FOUND_MOUNT);
+                /* Just to achieve device name uniqueness. Note that the suppresion of the duplicate
+                 * processing is merely an optimization, hence in case of OOM (unlikely) we'll just process
+                 * it twice. */
+                if (set_put_strdup_full(&devices, &path_hash_ops_free, device))
```

Comment:
please do an explicit != 0 comparison.

our coding style says we should only rely on C's downgrade-to-bool feature if values are either boolean/boolean-like or are pointers. For numeric/arithmetic values please do explicit numeric comparisons, i.e. use != 0

Reply:
OK， I'll change it as requested.